modules/streams/arrow-streams/src/test/kotlin/arrow/streams/internal/FreeCTest.kt

```diff
@@ -0,0 +1,180 @@
+        FreeC.liftF(s.right())
+          .run(Either.monadError()) == Right(Some(s))
+      }
+    }
+
+    "Running a Bind value" {
+      forAll(Gen.string()) { s ->
+        FreeC.Bind(FreeC.pure("")) { _ -> FreeC.pure<EitherPartialOf<Throwable>, String>(s) }
+          .run(Either.monadError()) == Right(Some(s))
+      }
+    }
+
+    "Running an suspended value"{
+      forAll(Gen.string()) { s ->
+        FreeC.defer() { FreeC.pure<EitherPartialOf<Throwable>, String>(s) }
+          .run(Either.monadError()) == Right(Some(s))
+      }
+    }
+
+    "map" {
+      forAll(Gen.string(), genFunctionAToB<String, String>(Gen.string())) { s, f ->
+        FreeC.pure<EitherPartialOf<Throwable>, String>(s)
+          .map(f)
+          .run(Either.monadError()) == Right(Some(f(s)))
+      }
+    }
+
+    "flatMap"{
+      forAll(Gen.string(), genFunctionAToB<String, String>(Gen.string())) { s, f ->
+        FreeC.pure<EitherPartialOf<Throwable>, String>(s)
+          .flatMap { FreeC.pure<EitherPartialOf<Throwable>, String>(f(it)) }
+          .run(Either.monadError()) == Right(Some(f(s)))
+      }
+    }
+
+    "asHandler"{
+      forAll(Gen.string(), genThrowable()) { s, t ->
+        FreeC.pure<EitherPartialOf<Throwable>, String>(s)
+          .asHandler(t)
+          .run(Either.monadError()) == Left(t)
+      }
+    }
+
+    "translate pure value"{
+      forAll(Gen.string()) { s ->
+        FreeC.pure<EitherPartialOf<Throwable>, String>(s)
+          .translate(EitherToTry)
+          .run(Try.monadError()) == Success(Some(s))
+      }
+    }
+
+    "translate fail value"{
+      forAll(genThrowable()) { t ->
+        FreeC.raiseError<EitherPartialOf<Throwable>, String>(t)
+          .translate(EitherToTry)
+          .run(Try.monadError()) == Failure(t)
+      }
+    }
+
+    "Running a interrupted value without errors" {
+      FreeC.interrupted<EitherPartialOf<Throwable>, String, Token>(Token(), None)
+        .translate(EitherToTry)
+        .run(Try.monadError()) == Success(None)
+    }
+
+    "Running a interrupted value with errors" {
+      forAll(genThrowable()) { t ->
+        FreeC.interrupted<EitherPartialOf<Throwable>, String, Token>(Token(), t.some())
+          .translate(EitherToTry)
+          .run(Try.monadError()) == Failure(t)
+      }
+    }
+
+    "translate liftF value"{
+      forAll(Gen.string()) { s ->
+        FreeC.liftF<EitherPartialOf<Throwable>, String>(s.right())
+          .translate(EitherToTry)
+          .run(Try.monadError()) == Success(Some(s))
+      }
+    }
+
+    "Running a Bind value" {
+      forAll(Gen.string()) { s ->
+        FreeC.Bind(FreeC.pure("")) { _ -> FreeC.pure<EitherPartialOf<Throwable>, String>(s) }
+          .translate(EitherToTry)
+          .run(Try.monadError()) == Success(Some(s))
+      }
+    }
+
+  }
+}
+
+internal object EitherToTry : FunctionK<EitherPartialOf<Throwable>, ForTry> {
+  override fun <A> invoke(fa: Kind<EitherPartialOf<Throwable>, A>): Kind<ForTry, A> =
```

Comment:
We have `Try#toEither`. Perhaps this implementation can be exposed in the core as `Either<Throwable, A>.toTry()`

Reply:
Yeah sure. Btw should we add `FunctionK` instances like this one to `Arrow`? Or is there a ticket for this that I missed?

Reply:
@nomisRev they won't hurt, but it isn't a priority. Only the ones between Try/Either/Option and maybe ListK may be more interesting for now.

Reply:
Just a friendly reminder that in 0.8.0 the use of `ListK` and wrappers of one type arguments are rarely seen in user code unless they have polymorphic functions that expect values in the kind hierarchy.